util/src/main/java/tc/oc/pgm/util/menu/InventoryMenuManager.java

```diff
@@ -0,0 +1,151 @@
+package tc.oc.pgm.util.menu;
+
+import com.google.common.collect.Maps;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryCloseEvent;
+import org.bukkit.event.inventory.InventoryMoveItemEvent;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.plugin.Plugin;
+import tc.oc.pgm.util.inventory.InventoryUtils;
+
+public class InventoryMenuManager implements Listener {
+
+  private final Map<Inventory, InventoryMenu> inventoryMap;
+  private final Plugin plugin;
+
+  /** Creates a new inventory manager. */
+  public InventoryMenuManager(Plugin plugin) {
```

Comment:
Instead of making one of these for each `Match`, register it as a global singleton in `PGMPlugin`